File: 914-x-of-a-kind-in-a-deck-of-cards/914-x-of-a-kind-in-a-deck-of-cards.py
class Solution:
    def hasGroupsSizeX(self, deck: List[int]) -> bool:
        def gcd(a, b):
            while b:
                if a>b:
                    a,b=b,a
                b%=a
            return a
        
        dic={}
        for i in deck:
            if i not in dic:
                dic[i]=1
            else:
                dic[i]+=1
        res=dic[deck[0]]
        for v in dic.values():
            res = gcd(res, v)
        
        if res==1:
            return False
        else:
            return True

File: 914-x-of-a-kind-in-a-deck-of-cards/914-x-of-a-kind-in-a-deck-of-cards.cpp
class Solution {
public:
    int gcd(int a, int b){
        while(b){
            if(a>b){
                int temp=a;
                a=b;
                b=temp;
            }
            b=b%a;
        }
        return a;
    }
    bool hasGroupsSizeX(vector<int>& deck) {
        unordered_map<int,int>mp;
        for(int i: deck){
            mp[i]++;
        }
        int res=mp[deck[0]];
        for(auto it: mp){
            res = gcd(res, it.second);
        }
        return res==1? false: true;
    }
};